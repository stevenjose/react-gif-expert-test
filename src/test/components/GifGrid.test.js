import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from "enzyme/build";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

const category = 'Dragon'

describe('debe cargar completo', () => {
    test('prueba componente < GifGrid category={ category } />', ()=> {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={ category } />);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar item cuando se carga imagenes useFetchGifs', ()=>{
        useFetchGifs.mockReturnValue({
            data: [{
                'id': '1789',
                'title': 'jgla',
                'url': 'localhost/jgla.png'
            }],
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={ category } />);
        expect( wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(1);
    })
});