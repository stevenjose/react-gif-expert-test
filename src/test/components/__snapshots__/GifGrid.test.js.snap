// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe cargar completo debe mostrar item cuando se carga imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Dragon
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="1789"
      key="1789"
      title="jgla"
      url="localhost/jgla.png"
    />
  </div>
</Fragment>
`;

exports[`debe cargar completo prueba componente < GifGrid category={ category } /> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Dragon
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
